__tests__/Port.tests.js

```diff
@@ -0,0 +1,32 @@
+const Port = require('../src/Port.js');
+
+describe('Port', () => {
+  let port;
+  beforeEach(() => {
+    port = new Port('Dover');
+  });
+  it('A new port can be instantiated', () => {
+    expect(port).toBeInstanceOf(Object);
+  });
+  it('A new port has a name property', () => {
+    expect(port.name).toEqual('Dover');
+    expect(port.ships).toEqual([]);
+  });
+  it('A port can add a ship to it\'s ship array', () => {
+    const ship = jest.fn();
+    port.addShip(ship);
+    expect(port.ships).toContain(ship);
+  });
+  it('A port can remove a ship from it\'s ship array', () => {
+    const titanic = { name: 'Titanic' };
+    const queenMary = { name: 'Queen Mary' };
+    const ssGreatBritain = { name: 'SS Great Britain' };
+    port.addShip(titanic);
+    port.addShip(queenMary);
+    port.addShip(ssGreatBritain);
+    port.removeShip(queenMary);
+    expect(port.ships).toEqual([{ name: 'Titanic' }, { name: 'SS Great Britain' }]);
+    port.removeShip(titanic);
+    expect(port.ships).toEqual([{ name: 'SS Great Britain' }]);
```

Comment:
both cases delete the first element in the array, spice things up. add 5 items to array

delete first element,
delete one in the middle,
delete last,
try to delete a non-existent item

Reply:
when unit testing always exercise boundaries, because that's where code tends to be flaky.

Reply:
👍